File: backend/dotnet-webapi/src/WorldCup.Api/Dtos/BudgetResultDto.cs
namespace WorldCup.Api.Dtos;

public class BudgetResultDto
{
    public bool Feasible { get; set; }
    public OptimisedRouteDto? Route { get; set; }
    public CostBreakdownDto CostBreakdown { get; set; } = new();
    public List<string> CountriesVisited { get; set; } = new();
    public List<string> MissingCountries { get; set; } = new();
    public double? MinimumBudgetRequired { get; set; }
    public List<string> Suggestions { get; set; } = new();
}

File: backend/java-spring/src/main/java/com/unosquare/worldcup/dto/BudgetResultDTO.java
package com.unosquare.worldcup.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BudgetResultDTO {
    private Boolean feasible;
    private OptimisedRouteDTO route;
    private CostBreakdownDTO costBreakdown;
    private List<String> countriesVisited;
    private List<String> missingCountries;
    private Double minimumBudgetRequired;
    private List<String> suggestions;
}
